File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0, right = nums.length - 1;
        int ans = nums[0];

        while (left <= right) {
            int mid = left + (right - left) / 2;

            ans = Math.min(ans, nums[mid]);


            if (nums[mid] <= nums[right]) {
                right = mid - 1;
            } 

            else {
                left = mid + 1;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root :
            return None

        stack = [root]
        ans = root

        while stack:
            root = stack.pop()

            temp = root.right
            root.right = root.left
            root.left = temp

            if root.right:
                stack.append(root.right)

            if root.left:
                stack.append(root.left)

        return ans